lambda_src/drivers/destination_s3.py

```diff
@@ -107,9 +109,11 @@ def write(
     )
 
     if not prefix:
-        prefixed_filename = f'{DATA_FOLDER_NAME}/{batch_id}/row-{row_index}.data.json'
+        prefixed_filename = f'{batch_id}-row-{row_index}.data.json'
+    elif prefix.endswith('/'):
+        prefixed_filename = f'{prefix}{batch_id}-row-{row_index}.data.json'
     else:
-        prefixed_filename = f'{prefix}/{batch_id}/row-{row_index}.data.json'
+        prefixed_filename = f'{prefix}-{batch_id}-row-{row_index}.data.json'
```

Comment:
no need to separate `prefix` and `batch_id` on the users' behalf (they can do it if they want), i.e. we can collapse these two cases

Reply:
Sure, will remove remove that case.

Reply:
yes, I think these three can be collapsed into a single case